Try tidying up text decoration across the codebase (I know.... :'))

tellthemachines opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #99254

## Proposed Changes

This is a very naive first attempt at removing the global text-decoration rule that affects all anchor tags, and replacing it with scoped rules for areas where we're pretty sure we don't need underlines on links.

For now I just removed the base rule and re-added it inside the nav sidebar and the card component.

I've noticed right away that there are underlines missing in places where I think they should exist, such as the copy inside this modal:

<img width="706" alt="Screenshot 2025-03-14 at 4 47 07 pm" src="https://github.com/user-attachments/assets/8a312296-5f95-482e-8056-75e5cc5db19a" />

Looks like there are more targeted text-decoration rules that will also have to be removed 😬 

To be updated as more testing occurs!

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Look through all the Calypso screens and check whether there are underlines where there shouldn't be (and vice-versa)

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

tellthemachines committed:
Try tidying up text decoration across the codebase (I know.... :'))

jasmussen commented on client/assets/stylesheets/_main.scss:
```diff
@@ -259,9 +259,9 @@ th {
-a {
-	text-decoration: none;
-}
+// a {
```
No doubt on your radar, but of course we'll just want to delete this outright, when we move out from draft state.

tellthemachines replied:
Yes, of course! I'll go ahead and remove it.

jasmussen commented on client/my-sites/sidebar/style.scss:
```diff
@@ -155,6 +155,10 @@ $font-size: rem(14px);
 		min-height: calc(100vh - var(--masterbar-height));
 		box-sizing: border-box;
 
+		a {
```
This seems a smart way to proceed; remove the root rule, then apply it locally.

jasmussen commented:
Excellent work, this seems pragmatically a good path forward: remove the blanket rule, then scope it locally until such a time as we can perhaps remove it entirely in favor of component-level CSS.

I took a surface level look at this PR in quick spot testing, and things look mostly right and good. I noticed a few things. Here's stats in trunk:

![Screenshot 2025-03-14 at 09 14 44](https://github.com/user-attachments/assets/0ff7ab45-1458-49c0-a106-7d6522d677e4)

Here's with this PR:

![Screenshot 2025-03-14 at 09 15 22](https://github.com/user-attachments/assets/6d117a4d-e59f-4f62-8d7e-781aa6f00a9b)

To be clear: this shows why the PR is necessary. The in-paragraph hyperlink _should_ be underlined, so it fixes a bug here. But the "View more" link at the bottom should arguably be a tertiary button. May be worth a CC to @Automattic/jetpack for awareness. 

There's something off about the purchases screen that may need a temporary blanket rule, but it's worth looking at other instances that use the same componentry (tabbar, drilldown), just to see if there are additional instances:

![Screenshot 2025-03-14 at 09 15 43](https://github.com/user-attachments/assets/b1f3c2ac-528c-49d9-9dfb-a426f604e7ad)

The reader will likely need some work as well. Here's before:
![Screenshot 2025-03-14 at 09 17 21](https://github.com/user-attachments/assets/4041578d-c37a-42d1-9c29-32671529ed3d)
Here's after:

 
![Screenshot 2025-03-14 at 09 17 26](https://github.com/user-attachments/assets/0d3e6efb-3743-43d3-ba47-70dd05076919)

We need underlined links inside _the content_, and "MORE IN SPLICED PERSONALITY", but the username and site name on the left, as well as post title and date below, should not be underlined.

Perhaps because it's a package and I haven't fully absorbed how to compile/test those, I did not see underlines in help, despite needing it:

![Screenshot 2025-03-14 at 09 17 38](https://github.com/user-attachments/assets/2fbe5a5b-8290-4233-b5c0-df48620bf0a7)

Some links in profile will also eventualy benefit from underlines:
![Screenshot 2025-03-14 at 09 18 45](https://github.com/user-attachments/assets/c4500c01-59ff-4d0c-a974-acc5aea08dcf)

tellthemachines committed:
Remove base rule

tellthemachines committed:
Add rules where needed and remove from help center

tellthemachines commented:
Thanks for testing @jasmussen! I removed the unwanted underlines from the reader and a few tabs navs, and deleted the rule that was removing them from the help centre. Also removed underlines appearing in the masterbar links. 

TODO: 

* profile page
* replace those "view more" links in the stats with a button component?

* I found this "card" in Plugins that's not an actual card so didn't get the style update: (I assume it shouldn't be underlined?)

<img width="331" alt="Screenshot 2025-03-17 at 4 53 40 pm" src="https://github.com/user-attachments/assets/16076cc7-1ca8-4fc2-af89-9c7196d5b0c2" />

* In the help centre, should the table of contents have underlines or not?

<img width="373" alt="Screenshot 2025-03-17 at 4 48 22 pm" src="https://github.com/user-attachments/assets/b0787e2d-60ae-4646-a1f6-5005496afc97" />

jasmussen commented:
> I found this "card" in Plugins that's not an actual card so didn't get the style update: (I assume it shouldn't be underlined?)

They way those links are placed, arguably they should be undrelined. And I think that's an aspect of this work that's meaningful: it makes us make better decisions, design-time, about how these should be put together. For now, however, it's probably best to have it match trunk; I expect that overall section to get some separate redesign work from a design system perspective, so it will no doubt be revisited.

> In the help centre, should the table of contents have underlines or not?

Let's keep that sans-underlines for now, as I would expect the help center to visually move towards how wp.org is designed, which also doesn't underline TOC items ([example](https://wordpress.org/documentation/article/faq-i-make-changes-and-nothing-happens/)), as they are more clearly list items.

tellthemachines committed:
Remove underlines from link card and support table of contents

tellthemachines committed:
Fix links inside paragraphs inside cards

tellthemachines committed:
Remove underlines from vertical nav

jasmussen commented:
I'm happy to 👍 👍 this one, is it ready for review?

tellthemachines commented:
> I'm happy to 👍 👍 this one, is it ready for review?

Yes I've looked everywhere I could think and haven't found anything else looking off. Let's say it's ready 😄

jasmussen reviewed (approved):
Great first step, impactful one but tests safely in my overall smoketesting. Let's keep an eye out for any followups that might need to happen, I'm also happy to help there.

![Screenshot 2025-03-19 at 10 13 30](https://github.com/user-attachments/assets/307cfb49-6547-486c-a9ce-2b160bf7a88c)
![Screenshot 2025-03-19 at 10 13 44](https://github.com/user-attachments/assets/8a5a70d4-c130-4860-abeb-178919c626b6)
![Screenshot 2025-03-19 at 10 13 54](https://github.com/user-attachments/assets/41f1d8db-1e14-4f1e-b7e3-8544971a9dcc)
![Screenshot 2025-03-19 at 10 13 57](https://github.com/user-attachments/assets/8a7cb389-c394-44c2-accb-9efa7fb4dc61)

adamwoodnz commented:
The [support page](https://wordpress.com/help) results have underlines which aren't well styled. Each result is wrapped in an anchor tag, the styles of which propogate to the heading and paragraph tags within:

![Screenshot 2025-04-08 at 2 49 20 PM](https://github.com/user-attachments/assets/aee9bf40-e4e9-4a00-b078-c4fa025a3ed5)

I can't find an existing issue for this, is it known? @jasmussen your thoughts on styles please; should the underline simply be turned off for these card style links?

tellthemachines commented:
@adamwoodnz it's a shame those aren't actual card components, because the styles for those were fixed already. As it seems these are meant to be card-like, I don't think they should have the underlines. It's a quick fix, I can put up a PR for it.

tellthemachines commented:
Here: https://github.com/Automattic/wp-calypso/pull/102464

jasmussen commented:
Thanks both for the quick fix here, indeed the temporary solution was to remove the underlines there.

However I believe we have an opportunity to do better with support pages like that. For example ask: what is the role of https://wordpress.com/help vs. https://wordpress.com/support/ when we also have the help center? It seems like we can consolidate on the two latter in the future. CC: @matt-west and @fditrapani for instincts.

fditrapani commented:
Hey folks, there is a plan to retire `/help`. `/support` will need to continue to exist as it helps us from an SEO perspective and allow people to access our documentation while logged out. It also feeds the content into the Help Center. I could see the help center eventually evolving into a general AI assistant (for support but also for site building and content generation) but I think it'll take time to get there.
